meeting-notes/2016-12-01.md

```diff
@@ -0,0 +1,308 @@
+controversial, to “ABI Stable Node”, which is what is most important to users.
+
+@mhdawson: Tested some sizeable modules, perf is within reasonable limits. Next
+step is error handling. Also need more feedback from community.
+
+API is C, some VMs use C++. Goal is to ensure C API and ABI are stable. C++
+wrappers/helpers may not be as stable. We’d like to add some common C++-style
+stuff, like `NAN::New`.
+
+Completed:
+
+* Ported all NAN examples.
+* Ported leveldown and nanomsg.
+
+Next:
+
+* Error handling
+* Perf evaluation.
+
+At that point make available to community and seek further feedback.
+
+@mhdawson: A challenge is that now people can use anything in V8. We will need
+to constrain and support only a subset.
+
+That means this won’t work for 100% of modules, i.e. those which do significant
+work specific to V8.
+
+@jasnell: Are there things in Node which should also be part of this API?
+
+@mhdawson: Some libuv abstractions were already in NAN, learning from some of
+those.
+
+Buffer is another example.
+
+@aruneshchandra: Thinking of trying with native modules in CITGM. 
+
+@thealphanerd: There are 4-5 native modules now in CITGM, we could add more.
+
+@indexzero provided a list: <https://github.com/nodejs/abi-stable-node/issues/22>
+
+Could we spin up a CITGM instance for testing NAPI in particular?
+
+@thealphanerd: Some of the modules have unusual build steps, they’ll have to be
+investigated. Also CITGM can take a script instead of the default test runner.
+
+What are the next steps on the [EP](https://github.com/nodejs/node-eps/pull/20)?
+
+@mhdawson: Wanted to land as a draft, but still only a PR. We don’t have answers
+to all the questions so haven’t been able to get consensus.
+
+@rvagg: Need to work on landing as a Draft, which is an intended thing. Add
+approval of Node-EP to next week’s CTC agenda.
+
+@aruneshchandra: At what point can we land this in the main repo?
+
+@rvagg: Could land as experimental.
+
+@jasnell: Once we land as experimental will there be a way to back out? See
+v8_inspector as an example.
+
+Discussion:
+* Landing as experimental gives opportunity for more feedback.
+* Would be behind a configure (build-time) flag, not as discoverable as a
+  runtime flag, so less risk of premature dependence.
+
+@aruneshchandra: Should we backport to previous versions to prove usefulness?
+
+@mhdawson: At least 2 LTS's.
+
+And Node-ChakraCore.
+
+First cuts for v8.x to be at end of March, be ready to bring this in behind a
+flag at that time.
+
+@rvagg: Need to ensure CTC is in the loop. Can perhaps achieve that through
+review of Node-EP.
+
+@jasnell: What do we need to achieve to consider NAPI feature-complete? All of
+V8 API?
+
+@aruneshchandra: No, not all of V8, but all that is used by native modules.
+
+@rvagg: But modules may use a lot of those APIs (e.g. garbage collection).
+
+@mhdawson: We’re assuming that’s a low number and we won’t address those.
```

Comment:
+@mhdawson: We’re assuming that’s a low number and we won’t address those in the first version.